### ISSUE TITLE
redirection convenience

### ISSUE THREAD

**Issue by casperdcl** [tqdm/tqdm#857]
- add `writelines` to `contrib.DummyTqdmFile`
- add `std_out_err_redirect_tqdm` to `contrib`
- continues #713
- related: #855

**Comment by KOLANICH** [comment]
IMHO - classes used for redirection should be moved into a separate package.

What thing I deeply dislike about the redirection is that it modifies the global state. So we need a refcounting self-destructing singleton to properly support reentrying and multithreading.

**Comment by casperdcl** [comment]
I agree - and `tqdm.contrib` is a separate subpackage (part of #855)

**Comment by casperdcl** [comment]
Btw the new way does wrap everything (incl. `flush()` and any other members)

**Comment by KOLANICH** [comment]
>tqdm.contrib is a separate subpackage

I mean a separate repo and a pip package.
